source/adiabatic_conditions/initial_profile.cc

```diff
@@ -72,11 +72,14 @@ namespace aspect
           in.position[0] = representative_point;
           in.temperature[0] = temperatures[i-1];
           in.pressure[0] = pressures[i-1];
+          in.velocity[0] = Tensor <1,dim> ();
 
           for (unsigned int c=0; c<n_compositional_fields; ++c)
             in.composition[0][c] = this->get_compositional_initial_conditions().initial_composition(representative_point, c);
 
           in.strain_rate.resize(0); // we do not need the viscosity
+          in.cell = this->get_dof_handler().end(); // and do not know the cell index
```

Comment:
Yes, your first form will work. It even means that wherever you want to test, you don't even have to have access to the `dof_handler`.